Assets/Scripts/Tags/Hazard.cs

```diff
@@ -1,45 +1,41 @@
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public class Hazard : MonoBehaviour {
+    [Tooltip("Defines how high and low this object will bob up and down, measured in world-space units.")]
+    [SerializeField] private float amplitude = 0.2f;
 
-    [SerializeField] private float verticalSpeed;
-    [SerializeField] private float amplitude;
+    [Tooltip("The rate at which this object will rotate over time, measured in degrees per second.")]
+    [FormerlySerializedAs("rotSpeed")]
+    [SerializeField] private float angularSpeed = 60;
 
-    [SerializeField] private float rotSpeed;
+    [Tooltip("Defines the range within which a vertical speed will be randomly chosen for this object.\n\n" +
+        "This object's vertical speed describes the frequency of the up and down bobbing motion.")]
+    [SerializeField] private Vector2 verticalSpeedRange = new Vector2(2, 3);
```

Comment:
Notice how I use a Vector2 to describe a min value (x) and max value (y).
The fact that I use the x as the min, and y as the max is arbitrary.
'Made up!

But it's nice to group the values together, and adding the suffix "Range" to the variable name makes it obvious enough to the user (in my opinion!) that this is a min-max range.

Later in the script, I use `Random.Range(verticalSpeedRange.x, verticalSpeedRange.y)` to actually pick a random value between those min-max values (ON START), and use it for the rest of the game.

Reply:
That makes sense! I like it.